src/Symfony/Component/Validator/Constraints/ExpressionValidator.php

```diff
@@ -46,6 +47,10 @@ public function validate($value, Constraint $constraint)
         $variables['value'] = $value;
         $variables['this'] = $this->context->getObject();
 
+        if (strlen($constraint->dataPath) > 0) {
```

Comment:
What about allowing the `$dataPath` option to also be a `PropertyPath` instance?